### ISSUE TITLE
在复现ETTh1时出现的欠拟合早停问题

### ISSUE THREAD

**Issue by Vangocjy** [decisionintelligence/DUET#19]
尊敬的作者：
  你好，
  我在复现论文时发现在运行Readme示例脚本时发现会有欠拟合早停的情况仅在前几个epoch对于验证集的loss进行收敛，例如：运行如下脚本
`python ./scripts/run_benchmark.py --config-path "rolling_forecast_config.json" --data-name-list "ETTh1.csv" --strategy-args '{"horizon": 720}' --model-name "duet.DUET" --model-hyper-params '{"CI": 1, "batch_size": 32, "d_ff": 512, "d_model": 512, "dropout": 0.2, "e_layers": 2, "factor": 3, "fc_dropout": 0.1, "horizon": 720, "k": 2, "loss": "MAE", "lr": 0.0005, "lradj": "type1", "n_heads": 1, "norm": true, "num_epochs": 100, "num_experts": 4, "patch_len": 48, "patience": 5, "seq_len": 512}' --deterministic "full" --gpus 0 --num-workers 1 --timeout 60000 --save-path "ETTh1/DUET`
输出：
`Total trainable parameters: 5957871
Validation loss decreased (inf --> 0.847442).  Saving model ...
Updating learning rate to 0.0005
Validation loss decreased (0.847442 --> 0.837697).  Saving model ...
Updating learning rate to 0.00025
EarlyStopping counter: 1 out of 5
Updating learning rate to 0.000125
EarlyStopping counter: 2 out of 5
Updating learning rate to 6.25e-05
EarlyStopping counter: 3 out of 5
Updating learning rate to 3.125e-05
EarlyStopping counter: 4 out of 5
Updating learning rate to 1.5625e-05
EarlyStopping counter: 5 out of 5
scheduling DUET: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [02:34<00:00, 154.17s/it]
collecting DUET: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<?, ?it/s]
`
我发现仅在两个epoch后就早停了，我尝试了修改了配置文件中的一些超参数但是还是于事无补，请问作者该如何解决

**Comment by qiu69** [comment]
您好：
       感谢您的关注与提问！
       几个epoch停止是模型自己决定的，我们设定的epoch为100，但是patience为5（代表如果验证loss，连续5次没有下降，模型就停止训练了）。不同数据集或者不同的模型参数，会导致模型训练的轮次不一样。
       您可以加我微信进一步交流：qxf780310
邱翔飞
华东师范大学
